app/models/user.rb

```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  has_secure_password
+  validates_confirmation_of :password
+  validates :password, :presence => true, :confirmation => true
```

Comment:
Nice addition of the `:confirmation => true` just read the docs and I like that idea. Additionally, nicely tied into the views. 

We should probably add `uniqueness: true` validations on email right?

Reply:
good idea. i'll add it now